File: ChatAppAPI/Client/src/app/security/app-user-auth.ts
import { UserAuthBase } from '../shared/security/user-auth-base';

export class AppUserAuth extends UserAuthBase {
  canAccessChat: boolean = false;

  init(): void {
    super.init();

    this.canAccessChat = false;
  }

  getValueOfProperty(obj: any, key: string): boolean {
    let ret = obj[key];
    return ret;
  }
}

File: Data/Entities/AppUserAuth.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Data.Entities
{
    public partial class AppUserAuth : UserAuthBase
    {
        public AppUserAuth() : base()
        {
            CanAccessChat = false;
        }

        public bool CanAccessChat { get; set; }
    }
}
